### PR TITLE
Version tests

### PR DESCRIPTION
**Pull request by mmirecki**
Check if the plugins runs correctly and returns the appropriate result based on the cniVersion received.

### CODE REVIEW

**Comment by mmirecki** [comment]
@killianmuldoon @Eoghan1232 could you please take a look?

**Review by killianmuldoon — COMMENTED** [review]
@mmirecki I'm not working on bond cni any more so don't have the bandwidth for reviews. Could you instead pull in the appropriate maintainers? 

Is there an owners file or somewhere where I'm listed as an owner here? If so I can see if I can get myself removed from it.

**Comment by mmirecki** [comment]
> @mmirecki I'm not working on bond cni any more so don't have the bandwidth for reviews. Could you instead pull in the appropriate maintainers?
> 
> Is there an owners file or somewhere where I'm listed as an owner here? If so I can see if I can get myself removed from it.

No owners file for this repo, and the privileges on a repo are not public. You were assigned as a reviewer by github

**Comment by killianmuldoon** [comment]
I created a PR on the community repo for NPWG (but that doesn't include bond CNI. I'm guessing its down to the group admins to take care of it - maybe something to bring up at the next community meeting.

It would be helpful if you could stop pinging me directly though - I'm not watching the repo, but I do get those pings 😆

File: bond/bond_test.go [hunk 1]
```diff
@@ -127,4 +129,100 @@ var _ = Describe("tuning plugin", func() {
 		})
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	It("verifies the plugin returns correct results for CNI versions 0.3.0, 0.3.1, 0.4.0", func() {
```
Comment (Eoghan1232) [inline bond/bond_test.go:133]: Could we just combine these 3 tests into 1? 
Seems only difference is the CNI version?

Perhaps create a list containing all the versions and loop through this?
WDYT?

Comment (mmirecki) [reply bond/bond_test.go:133]: The difference is the Results version we expect:
https://github.com/k8snetworkplumbingwg/bond-cni/pull/36/files#diff-663523154c95a1a2ed183e0a982eb928029e2dc37b6504dd680f38f277ef6663R150
https://github.com/k8snetworkplumbingwg/bond-cni/pull/36/files#diff-663523154c95a1a2ed183e0a982eb928029e2dc37b6504dd680f38f277ef6663R182
https://github.com/k8snetworkplumbingwg/bond-cni/pull/36/files#diff-663523154c95a1a2ed183e0a982eb928029e2dc37b6504dd680f38f277ef6663R214

Also the format of the results for 0.2/0.1 is completely different.

Comment (Eoghan1232) [reply bond/bond_test.go:133]: Okay I understand.

**Review by Eoghan1232 — APPROVED** [review]

File: bond/bond_test.go [hunk 2]
```diff
@@ -127,4 +129,100 @@ var _ = Describe("tuning plugin", func() {
...
+			args := &skel.CmdArgs{
+				ContainerID: "dummy",
+				Netns:       podNS.Path(),
+				IfName:      IfName,
+				StdinData:   []byte(fmt.Sprintf(Config, version)),
+			}
+			err := podNS.Do(func(ns.NetNS) error {
+				defer GinkgoRecover()
+
+				By("creating the plugin")
```
Comment (eoghanlawless) [inline bond/bond_test.go:144]: Better logging
```suggestion
				By("adding plugin " + version)
```

**Review by eoghanlawless — APPROVED** [review]

**Pull request merged**